### PR TITLE
Added Flask extensions for debugging and profiling

### PR DESCRIPTION
**Pull request by Dragneel7**
1.) Flask-Profiler: with admin authentication, keeps statistics of no. of user visits on a given URL, average visits in a given period, average time taken.
2.) Apache Bench: automated requests generation for a given URL.
3.) Line-Profiling: Line-by-Line profiling of the code to find out the bottlenecks during queries.
4.) Sentry-integration.

### CODE REVIEW

**Comment by Dragneel7** [comment]
@cchristelis @pierrealixt Please have a look at this PR.
@timlinux Had given me a sentry DSN key and registered the app in kartoza team. I have not added the key in the PR.

**Comment by Dragneel7** [comment]
Flask-Proflier(Admin)

deepinscreenshot_20180612155855


Line-Profiler

line_profiler

File: flask_project/runserver.py [hunk 1]
```diff
@@ -1,3 +1,3 @@
-from app import osm_app
+from app import *
```
Comment (pierrealixt) [inline flask_project/runserver.py]: why?

File: flask_project/campaign_manager/reminderRedisClient.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from redis import Redis
```
Comment (pierrealixt) [inline flask_project/campaign_manager/reminderRedisClient.py]: redis?

**Comment by pierrealixt** [comment]
It would be great to profile SQL requests.

capture d ecran 2018-07-12 a 10 35 37

**Comment by pierrealixt** [comment]
I got error when creating a campaign, is it normal?
capture d ecran 2018-07-12 a 10 42 53

**Comment by Dragneel7** [comment]
@pierrealixt I'll look into how to configure SQLAlchemy, We have used SQL-Alchemy but this configuration seems for flask-SQLAlchemy.
I am not sure if this will possible but I'll give it a look.

**Comment by Dragneel7** [comment]
@pierrealixt I wasn't able to reproduce the built-in error.

**Pull request merged**